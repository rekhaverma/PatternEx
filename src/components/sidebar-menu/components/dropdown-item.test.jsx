import React from 'react';
import DropdownItem from './dropdown-item.component';

const itemsList = [
  {
    icon: 'icon-test1',
    id: 'test1',
    label: 'Test1',
    location: '/test',
    params: ['start_time', 'end_time'],
    type: 'link',
  },
  {
    icon: 'icon-test2',
    id: 'test2',
    label: 'Test2',
    location: '/test',
    params: ['start_time', 'end_time'],
    type: 'link',
  },
];

describe('<DropdownItem />', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      active: true,
      activeDropdown: true,
      className: 'dropdownItem__test',
      disabled: false,
      id: 'testItem',
      icon: 'test-icon',
      items: itemsList,
      label: 'Test Label',
      onClick: jest.fn(),
      onDropdownClick: jest.fn(),
    };
    wrapper = shallow(<DropdownItem {...props} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

