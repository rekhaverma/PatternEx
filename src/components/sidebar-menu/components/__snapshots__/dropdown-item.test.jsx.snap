// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownItem /> should render correctly 1`] = `
<div
  className="dropdownItem__test--dropdown +active"
  onClick={[Function]}
>
  <div
    className="dropdownItem__test__dropdown-item +active"
  >
    <span
      className="dropdownItem__test__icon test-icon"
    />
    <div
      className="dropdownItem__test__dropdown-title"
    >
      <p
        className="dropdownItem__test__label"
      >
        Test Label
      </p>
      <span
        className="icon-chevron-down dropdownItem__test__arrow-icon +rotate"
      />
    </div>
  </div>
  <div
    className="dropdownItem__test__dropdown-list +no-items-2"
  >
    <Item
      active={false}
      activeDropdown={false}
      className="sidebarMenu__item"
      disabled={false}
      icon="icon-test1"
      id="test1"
      items={Array []}
      key="test1"
      label="Test1"
      location="/test"
      onClick={[Function]}
      onDropdownClick={[Function]}
      params={
        Array [
          "start_time",
          "end_time",
        ]
      }
      query={Object {}}
      style={Object {}}
      type="link"
    />
    <Item
      active={false}
      activeDropdown={false}
      className="sidebarMenu__item"
      disabled={false}
      icon="icon-test2"
      id="test2"
      items={Array []}
      key="test2"
      label="Test2"
      location="/test"
      onClick={[Function]}
      onDropdownClick={[Function]}
      params={
        Array [
          "start_time",
          "end_time",
        ]
      }
      query={Object {}}
      style={Object {}}
      type="link"
    />
  </div>
</div>
`;
